src/treecons.ml

```diff
@@ -0,0 +1,131 @@
+(** Tree constructors: a simpler representation of terms to be used by a
+    decision tree. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_mod : string list
+      (** Module name where the symbol of the constructor is defined. *)
+      ; c_sym : string
+      (** Symbol of the constructor. *)
+      ; c_ari : int
+      (** Arity of the considered constructor.  A same symbol representation
+          may generate several constructors with different arities. *) }
```

Comment:
We perhaps already discussed this but I cannot remember. Could you please give an example where we would have the same symbol with different arities?

Reply:
You can look at `tests/OK/arity_diff.lp`

Reply:
Please give a link to this file.